docs/daemon/configuration/configuration.mdx

```diff
@@ -0,0 +1,359 @@
+
+- `timeout_prevote`: This variable sets the time to wait after receiving +2/3 prevotes for "anything". Default timeout is 1s.
+
+- `timeout_precommit`: This variable sets the time to wait after receiving +2/3 precommits for "anything". Default timeout is 1s.
+
+- `timeout_commit`: This variable sets the time to wait after committing a block before starting on the new height. Default timeout is 1s.
+
+### p2p
+This subsection describes the configuration to manage the p2p service handling peer connectivity between the nodes in the blockchain network.
+
+- `listen_addr`: This variable sets the address to listen for incoming connections. Default listening address is `tcp://0.0.0.0:26656`.
+
+- `external_address`: This variable sets the address to advertise to peers for them to dial. If empty, it will use the same port as the listening address, and will introspect on the listener.
+
+- `persistent_peers`: This variable sets the comma-separated list of nodes to connect to and maintain persistent connections. This is a required field for a new node to connect to the existing nodes in the Kwil network.
+
+- `addr_book_strict`: Toggle for strict address routability rules.
+
+- `max_num_inbound_peers`: This variable sets the Maximum number of inbound peers.
+
+- `max_num_outbound_peers`: This variable sets the Maximum number of outbound peers to connect to, excluding persistent peers.
+
+- `unconditional_peer_ids`: This variable sets the List of node IDs for which connections will be (re)established irrespective of the max connections.
+
+- `allow_duplicate_ip`: This variable sets the Toggle to disable guard against peers connecting from the same ip address.
+
+### mempool
+This subsection describes the configuration to manage blockchain's transaction mempool.
+
+- `size`: This variable sets the maximum number of transactions in the mempool.
+
+- `max_txs_bytes`: This variable sets the limit on the total size of all transactions in the mempool.
+
+- `cache_size`: This variable sets the size of the cache used to filter transactions seen earlier.
+
+This is the structure in which these configs must appear in the config.toml file. Refer to the sample [config.toml](./configuration.mdx#sample-configtoml) file for reference.
+
+## Config Override
+All the configs in `config.toml` can be overwritten either using the environment variables or the command line flags.  
+
+The order of precedence: `flags > env variables > config.toml > defaults`. 
+
+These configs can be translated into env variables or flags using the below rules: 
+
+
+:::note
+``` title=config.toml
+[section]
+   field1 = val1
+[section.subsection]
+   field2 = val2
+```
+The flag and env variables for field1 & field2 are as follows:
+```
+All environment variables corresponding to the configs are prefixed by "KWILD_"
+Env variable:
+- KWILD_SECTION_FIELD1
+- KWILD_SECTION_SUBSECTION_FIELD2
+
+Flags:
+- section.field1
+- section.subsection.field2
+```
+:::
```

Comment:
Can you better explain this section?  To someone who isn't familiar with this, it probably won't make any sense

Reply:
Perhaps rephrase and use a table:

<pre>
:::note
```toml title=config.toml
[section]
   field1 = val1
[section.subsection]
   field2 = val2
```

The corresponding command line flags and environment variables for `field1` and
`field2` are as follows:

| Flag                          | Environment Variable              |
|-------------------------------|-----------------------------------|
| `--section.field1`            | `KWILD_SECTION_FIELD1`            |
| `--section.subsection.field2` | `KWILD_SECTION_SUBSECTION_FIELD2` |

:::
</pre>

Renders like:

![image](https://github.com/kwilteam/docs/assets/140431406/a895a9c7-67ae-4fb8-888c-fc88c510b060)